wfe/web-front-end.go

```diff
@@ -1182,6 +1191,76 @@ func (wfe *WebFrontEndImpl) BuildID(response http.ResponseWriter, request *http.
 	}
 }
 
+// Options responds to an HTTP OPTIONS request.
+func (wfe *WebFrontEndImpl) Options(response http.ResponseWriter, request *http.Request, methodsOK map[string]bool) {
+	allowMethods := ""
+	for method := range methodsOK {
+		if allowMethods != "" {
+			allowMethods += ", "
```

Comment:
This should use strings.Join, but even better, the call to strings.Join in HandleFunc should be moved a little higher and saved to a var, then passed to Options if necessary. That way we can de-duplicate that join logic somewhat.